core/src/test/java/com/linecorp/armeria/common/RequestContextTest.java

```diff
@@ -176,6 +178,21 @@ public void makeContextAwareChannelFutureListener() {
     }
 
     @Test
+    public void makeContextAwareCompletableFuture() throws Exception {
+        RequestContext context = createContext();
+        CompletableFuture<String> future = context.makeContextAware(
+                CompletableFuture.supplyAsync(() -> "success", ImmediateEventExecutor.INSTANCE));
```

Comment:
I wonder what will happen if a real Executor such as a ThreadPoolExecutor is specified here. Could you add a test case for that as well?